### ISSUE TITLE
AMD GPU driver crash XR application on Windows 10

### ISSUE THREAD

**Issue by Cyberrebell** [godotengine/godot#61787]
### Godot version

4.0 alpha9

### System information

Windows 10, Vulkan, AMD RX6600 (22.5.1 -> latest installed by Adrenalin)

### Issue description

I created an XR application on Arch Linux using v4.0 Alpha 9 (which worked fine on Linux). Exported as .exe to test on Windows 10 and got a AMD driver crash (black screen, all Adrenalin Settings back to defaults...).

So I tried to reproduce it by creating a new test application on Windows:
1. alpha 9 non XR -> works fine
2. alpha 9 XR -> crash
3. alpha 5 XR -> works fine
4. alpha 6 (editor crashed when opening a project)
5. alpha 7 XR -> crash

I'm using an MSI AMD RX6600 with the default driver I get from Adrenalin on my Valve Index.

This is my CMD-Output I get when running the app from the godot editor:
```
OpenXR: Running on OpenXR runtime:  SteamVR/OpenXR   0.1.0
Godot Engine v4.0.alpha9.official.fc18891db - https://godotengine.org
OpenXR: XrGraphicsRequirementsVulkan2KHR:
 - minApiVersionSupported:  1.0.0
 - maxApiVersionSupported:  1.2.0
Vulkan API 1.2.0 - Using Vulkan Device #0: AMD - AMD Radeon RX 6600

OpenXR: Found supported reference space  XR_REFERENCE_SPACE_TYPE_VIEW
OpenXR: Found supported reference space  XR_REFERENCE_SPACE_TYPE_LOCAL
OpenXR: Found supported reference space  XR_REFERENCE_SPACE_TYPE_STAGE
OpenXR: Found supported swapchain format  VK_FORMAT_R8G8B8A8_SRGB
OpenXR: Found supported swapchain format  VK_FORMAT_B8G8R8A8_SRGB
OpenXR: Found supported swapchain format  VK_FORMAT_R32G32B32A32_SFLOAT
OpenXR: Found supported swapchain format  VK_FORMAT_R32G32B32_SFLOAT
OpenXR: Found supported swapchain format  VK_FORMAT_R16G16B16A16_SFLOAT
OpenXR: Found supported swapchain format  VK_FORMAT_D32_SFLOAT
OpenXR: Found supported swapchain format  VK_FORMAT_D16_UNORM
OpenXR: Found supported swapchain format  VK_FORMAT_D24_UNORM_S8_UINT
OpenXR: Found supported swapchain format  VK_FORMAT_D32_SFLOAT_S8_UINT
Using swap chain format: VK_FORMAT_R8G8B8A8_SRGB
ERROR: Vulkan: Did not create swapchain successfully.
   at: (drivers/vulkan/vulkan_contextO.pceppn:X1R9:5 9F)a
iled tEoR lRoOcRa:t eC ovnideiwt isopna c"ee rirn"  pilsa yt rsupea.c e R[e tXuRr_nEiRnRgO:R _ETRIRM_EC_AINNTV_ACLIRDE A]T
E
   at: swap_buffers (drivers/vulkan/vuElRkRaOnR_:co nVtuelxkta.nc:p pD:2i0d36 n)o
t create swapchain successfully.
   at: (drivers/vulkan/vulkan_context.cpp:1959)
ERROR: Condition "err" is true. Returning: ERR_CANT_CREATE
   at: swap_buffers (drivers/vulkan/vulkan_context.cpp:2036)
ERROR: vkBeginCommandBuffer failed with error -4.
   at: (drivers/vulkan/rendering_device_vulkan.cpp:8701)
ERROR: CEoRnRdOiRti:o nC o"nedrirt"i oins  "terrure".  iRse tturrunei.n gR:e tERuRr_nCiAnNgT:_ CERRERA_TCEA
NT  _ CaRtE:A T_Eu
pd a t aet_:s wa_pu_pcdhaatien_ s(wdapr_icvhearisn/ v(udlrkiavne/rvsu/lvkualnk_acno/nvtuelxkta.ncp_pc:o1nt6e6x3t).
cpp:1663)
ERROR: Condition "err" is true. Returning: ERR_CANT_CREATE
   at: _update_swap_chain (drivers/vulkan/vulkan_context.cpp:1663)
EERRRROORR::  CCoonnddiittiioonn  ""eerrrr""  iiss  ttrruuee..  RReettuurrnniinngg::  EERRRR__CCAANNTT__CCRREEAATTEE

      aatt::  __uuppddaattee__sswwaapp__cchhaaiinn  ((ddrriivveerrss//vvuullkkaann//vvuullkkaann__ccoonntteexxtt..ccpppp::11666633))

ERROR: Condition "err" is true. Returning: ERR_CANT_CREATE
   at: _update_swap_chain (drivers/vulkan/vulkan_context.cpp:1663)
```

### Steps to reproduce

- use an AMD GPU (RX 6600 does crash, RX 580 does not) on Windows 10
- use 4.0 alpha 9
- create a new XR project just with minimum components (1 mesh, 1 light, 1 XROrigin+XRCamera)
- enable XR in project settings (use desktop or mobile rendering backend, both will crash)
- run the project

### Minimal reproduction project

[test.zip](https://github.com/godotengine/godot/files/8874080/test.zip)

**Comment by Cyberrebell** [comment]
I did some more testing with different devices and software:
Quest 2 Airlink using SteamVR as OpenXR Server: crash
Quest 2 Airlink using the Oculus Rift Software as OpenXR Server: works fine
Quest 2 Airlink on a different Win10-PC with an Nvidia-GPU using SteamVR as OpenXR Server: works fine

So the crash seem to be specific to the combination:
Windows(10)+SteamVR+AMD-GPU+Godot4.0-Alpha7+
No matter which VR-Headset.

So would be nice if anyone who has an AMD-GPU and a VR Headset could try to reproduce it.

**Comment by BastiaanOlij** [comment]
Don't worry about alpha 6, we had a build where changes were introduced that just broke OpenXR all together.

I don't have AMD hardware to test so that makes things a little tricky. it would be interesting to know if this is specific SteamVR + Oculus Link or just streamVR. 

It's getting stuck on creating the swapchains which is really weird, it seems to be getting the swapchain formats just fine but then on creation it errors out. This aborts the whole startup of Godot and indeed I've found that Godot doesn't always exit cleanly in this condition.

**Comment by Cyberrebell** [comment]
> I don't have AMD hardware to test so that makes things a little tricky. it would be interesting to know if this is specific SteamVR + Oculus Link or just streamVR.
As mentioned in the initial post it does crash with my Valve Index. In this case I don't use any Oculus software but just SteamVR.

A friend with a AMD RX 580 tried my test application and it did not crash.
His VR setup: Quest 2 Airlink using SteamVR as OpenXR Server (also on Win 10)
Also he is not using the AMD Adrenalin driver but the driver that is automatically installed by Windows Update.

To test this too I installed the Windows Update default driver on my PC (which installs a version from jan 2022) but it does crash too. (Adrenalin installs a driver version from april)

So for now it looks like it only happens on the latest AMD GPUs... or just on my system... I'll try to find someone else with 6000 series GPU and a VR headset.

**Comment by Cyberrebell** [comment]
For further testing I compiled Godot myself on Windows using Visual Studio and this build works fine while the official build does still crash the AMD driver.
I tried the latest master version of Godot and the commit hash mentioned in the official [Alpha 9](https://downloads.tuxfamily.org/godotengine/4.0/alpha9/README.txt) build. They are both working.

So maybe it's a strange hardware specific mono (or mingw64) bug caused by a version update on the build server between  2022-Mar-24 and 2022-Apr-27.
Am I right that the official alpha builds are created using [this script](https://github.com/godotengine/godot-build-scripts/blob/godot4/build-windows/build.sh) inside [this docker image](https://github.com/godotengine/build-containers/blob/main/Dockerfile.windows)?

I could try to build it myself and maybe the bug doesn't occur in the latest mono version anymore.
It would also be helpful for me to know which exact mono version is currently used for official builds so I can replicate the bug and try to fix it by using a later version.

**Comment by Cyberrebell** [comment]
I got the build process working on my machine and found [the breaking commit](https://github.com/godotengine/godot/pull/59524/files#diff-0d547c4e6687e653e54df1623d33ab291396cca8c6265101c211a028e610b1e9).
I updated vk_mem_alloc to the latest version and it fixed the issue for me on Windows.

I'll do some testing in the upcoming days to be sure that fixing one platform doesn't break another platform.
